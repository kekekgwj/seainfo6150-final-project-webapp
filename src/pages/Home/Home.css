.home-wrapper {
    color: #111;
    width: 800px;
    margin: 0 auto;
}