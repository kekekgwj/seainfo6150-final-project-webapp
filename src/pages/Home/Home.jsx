import React, { useEffect, useState } from "react";
import "./Home.css";
import ReviewList from "../../components/ReviewList/ReviewList";
import { getUserReview } from "../../services";
import { errMsgs } from "../../base";

const Home = () => {
    const [data, setData] = useState({});
    const [status, setStatus] = useState('');

    useEffect(() => {
        getUserReview()
        .then( res => {
            setData(res);
        })
        .catch( err => {
            setStatus(errMsgs[err.error] || err.error);
        }
        )
    },[])

    return (
        <div className="home-wrapper">   
            <div className="home-review-list-wrapper">
                <h2 className="home-review-list-title">My Reviews · · · · · ·</h2>
                <div className="home-error">
                        <span className="error-message">{status}</span>
                </div>
                <ReviewList className="home-review-list" reviewData= {data}/>
            </div>

        </div>)
} 


export default Home;