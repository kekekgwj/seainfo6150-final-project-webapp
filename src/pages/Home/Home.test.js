import React from "react";
import Home from "./Home";

describe("Home page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Home/>);
    expect(container).toMatchSnapshot();
  });
});