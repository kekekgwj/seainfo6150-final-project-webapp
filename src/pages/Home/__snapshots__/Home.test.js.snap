// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page tests renders correctly 1`] = `
<div>
  <div
    class="home-wrapper"
  >
    <div
      class="home-review-list-wrapper"
    >
      <h2
        class="home-review-list-title"
      >
        My Reviews · · · · · ·
      </h2>
      <div
        class="home-error"
      >
        <span
          class="error-message"
        />
      </div>
      <div
        class="review-list-container"
      />
    </div>
  </div>
</div>
`;
